sqlmesh/core/model/definition.py

```diff
@@ -1867,8 +1904,8 @@ def _python_env(
     expressions = ensure_list(expressions)
     for expression in expressions:
         if not isinstance(expression, d.Jinja):
-            for macro_func_or_var in expression.find_all(d.MacroFunc, d.MacroVar):
-                if macro_func_or_var.__class__ is d.MacroFunc:
+            for macro_func_or_var in expression.find_all(d.MacroFunc, d.MacroVar, exp.Identifier):
+                if isinstance(macro_func_or_var, d.MacroFunc):
```

Comment:
sorry i thing i may have been wrong here, the problem is that macrosql and macro def are subclasses of macrofunc